internal/commands/root.go

```diff
@@ -10,15 +11,36 @@ import (
 	"github.com/spf13/pflag"
 )
 
+var (
+	completion string
+)
+
 // NewRootCmd returns the base root command.
 func NewRootCmd(use string, dockerCli command.Cli) *cobra.Command {
 	cmd := &cobra.Command{
 		Short:       "Docker Application",
 		Long:        `A tool to build and manage Docker Applications.`,
 		Use:         use,
 		Annotations: map[string]string{"experimentalCLI": "true"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			switch completion {
+			case "bash":
+				return cmd.GenBashCompletion(dockerCli.Out())
+			case "zsh":
+				return cmd.GenZshCompletion(dockerCli.Out())
+			case "":
+				// Actually unset
+				return nil
+			default:
+				return fmt.Errorf("%q is not a supported shell", completion)
+			}
+		},
 	}
 	addCommands(cmd, dockerCli)
+
+	cmd.Flags().StringVar(&completion, "completion", "", "Generates completion scripts for the specified shell (bash or zsh)")
+	_ = cmd.Flags().MarkHidden("completion")
```

Comment:
I think this error should be checked 🤔

Reply:
MarkHidden return an error is flag is not found, which is crazy as we just built it but StringVar doens't return the produced flag

Reply:
I know, the API is not great on that point, but I think we should still stick to the go mantra: check every error 😅